File: LeetCode/Medium/0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
import java.util.*;


class Solution {
        public int[] topKFrequent(int[] nums, int k) {

            Map<Integer, Integer> numsCountMap = new HashMap<>();
            for(int i = 0; i < nums.length; i++) {

                numsCountMap.put(nums[i], numsCountMap.getOrDefault(nums[i], 0)+1);
            }

            
            PriorityQueue<Map.Entry<Integer, Integer>> pq = new PriorityQueue<>(
                    (entry1, entry2) -> entry2.getValue().compareTo(entry1.getValue())
            );

            pq.addAll(numsCountMap.entrySet());

            int[] output = new int[k];
            for(int i = 0; i < k; i++) {
                output[i] = pq.poll().getKey();
            }

            return output;
        }
    }

File: LeetCode/Medium/0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
from typing import List
import collections
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        # 아래와 같이 초기화하면 딕셔너리의 기본 값을 0으로 설정함
        nums_count_map = collections.defaultdict(int)

        for num in nums:
            nums_count_map[num] += 1
        
        max_heap = [(-count, num) for num, count in nums_count_map.items()]
        heapq.heapify(max_heap)

        output = [heapq.heappop(max_heap)[1] for _ in range(k)]

        return output